### ISSUE TITLE
Cross-certify to prove ownership of endpoint keys

### ISSUE THREAD

**Issue by pospeselr** [blueprint-freespeech/gosling#17]
We should require that an introduction server also owns the provided endpoint server (by signing some message with the end point server's private key for instance).

@nmathewson Is the required configuration of client auth sufficient here? Connecting clients provide an x25519 public key for client auth (though I suppose we can't guarnatee clients won't reuse keys).

**Comment by nmathewson** [comment]
> Is the required configuration of client auth sufficient here?

I don't think so; client auth doesn't stop the introduction server from giving out the ID of an endpoint server that's actually controlled by a different user.  Not only might a client reuse keys, it might also already have a key with the specified endpoint.

**Comment by pospeselr** [comment]
The identity server should return this with the service_id from gosling_identity::send_response()

**Comment by pospeselr** [comment]
@nmathewson So the summary here is that I'll update the identity server hanshake to have the server return both the endpoint service id, as well as a signature for some proof signed with the ed25519 private key used to derive the endpoint service id (currently only the endpoint service is is returned).

Can this just be the *same* proof the client signed with their identity key as part of the handshake ( https://github.com/blueprint-freespeech/gosling/blob/main/docs/gosling_protocol/protocol.md#client-identity-proof-calculation-and-verification ), or should a new proof be generated, for example with a new `domain_seperator`, `server_service_id` (the returned endpoint service id), and `server_cookie`.
